Backend/README.md

```diff
@@ -86,3 +86,39 @@ Log in using:
 
 If you want to access the database, you are most likely prompted for a password. It is set to `postgres` by default.
 
+
+
+# Create Applicant Restful API
+
+```json
+{
+  "firstName": String,
+  "lastName": String,
+  "industries": {
+    "automotive": boolean,
+    "finance": boolean,
+    "commerce": boolean,
+    "pharma_Helthcare": boolean,
+    "public_Sector": boolean
+  },
+  "positions": {
+    "consultant_Business_Consultant": boolean,
+    "iT_Consultant_Informationsmanagement": boolean,
+    "iT_Consultant_Java_JEE": boolean,
+    "iT_Consultant_Data_Science": boolean,
+    "iT_Consultant_Artificial_Intelligence": boolean,
+    "internship_Working_Student": boolean,
+    "consultant_SAP": boolean
```

Comment:
So for every possible Position, you need to send a boolean to the server? Wouldn't it be better if you could simply send a string array to the server? Imagine we have 100 possible positions - then you also need to provide 100 values, even if the user only wants to work in 1 position... The same goes for the columns - you need n columns for n possible values. :/ Having a separate table (and using SQL Joins) could probably make things much easier :) What do you think?

Reply:
It depends on the data type, at the moment its a fixed data, so having it as boolean is the best solution for the DB.

Reply:
I will provide also a DB schema as a table so it will make it better to understand :)

Reply:
![DB-Schema](https://user-images.githubusercontent.com/56546243/83354188-a1304980-a357-11ea-93d6-7e78c8224f27.png)

Reply:
> So for every possible Position, you need to send a boolean to the server? Wouldn't it be better if you could simply send a string array to the server? Imagine we have 100 possible positions - then you also need to provide 100 values, even if the user only wants to work in 1 position... The same goes for the columns - you need n columns for n possible values. :/ Having a separate table (and using SQL Joins) could probably make things much easier :) What do you think?

@Nasser-Eddin-Nasser Even I feel it is better to have array of strings rather than a field for everything. The problem here is whenever a new field is added , backend API won't be backward compatible. You may need hack to support it backward compatible which may slow down the development. Also in backend will there gonna be any specific business logic related to these fields? I don't think at this moment u need to have specific handling for Automotive field or AI or DataScience in the backend.

Reply:
Hey, I also agree on it but, Anshaj means the Positions and Industry are fixed values (will not be changed), that's why I did it this way so that in our  DB we will store each position/Industry as a column with boolean values (less store capacity needed, and more efficient while having DB operations). 
In case we will store it as a string array values inside the DB we will have to insert more values in our DB table in one column (that's bad).

Another way to do is to map all the DB data to Objects in our BE Server and then make the whole operations we want and then update the data (more efficient while having DB operations because it will be made in Server and more capacity needed in the Server).

The problem is we still don't know how/who well change these values (Usually it should be updated/added in the BE and then by init new client/ client refresh, the values will be sent to the client (FE)  where it will be rendered there )! 

Sure we still can change it, it's not our last sprint :)

Reply:
@Bharathravi1995  @MBuchalik  Sure, thank you both for reviewing the Pull req.  :)

Reply:
![DB-Schema-2](https://user-images.githubusercontent.com/56546243/83425906-ad3b0a80-a42e-11ea-85f6-6d91ed4e5585.png)

Reply:
After having a meeting with @dalex135  and @vishwasanavatti  we decided to make DB schema "smarter", with a mapping function. the mapping will be done in the BE Server.  
please let me know if you have any questions or suggestions @MBuchalik  @Bharathravi1995

Reply:
In the "Industry" (and "Position") field of the first row, you wrote "1,2". Does that mean that you will store more than one value per field? What about normalizing it by creating a separate table (as suggested in my first comment)? Would this have some downsides? You wouldn't really need the mapping table then as well, since you could simply store arbitrary strings here.

I mean somthing like this:

Main table:
| ID | Phone Number | Email       |
|----|--------------|-------------|
| 1  | 12345678     | example@example.com |
| 2  | 987654321    | example@example.com |

"Positions" table:
| ID | Applicant ID (foreign key) | Position             |
|----|----------------------------|----------------------|
| 1  | 1                          | Some Position        |
| 2  | 1                          | Another Position     |
| 3  | 1                          | Yet another Position |
| 4  | 2                          | Some Position        |

What would be the downside of this?

Reply:
In case of having several values per field is the worst thing to do, but because some developers pointed out that Position & industry are not fixed (not in the requirement), so we discussed it together, and we decided to change our schema and to add a mapping layer.  
 In your example, the same position value will be stored several times on different users, in case we only have a few users (around 100) it will be fine otherwise we will have too big DB.
 What's the problem with the mapping structure?

Reply:
The mapping structure itself is ok. I am just concerned about the fact that you want to store multiple values in a single field. Or am I interpreting your diagram in the wrong way? (I don't know how Postgres works in detail. Maybe, this is just how you usually do it. I can only speak for other SQL databases and here, you always try to normalize your database. Maybe not to the highest normal form, but at least try not to violate 1NF.)

Btw, my example could also be updated to take your examples into account:

| ID | Applicant ID (foreign key) | Position (references the mapping table) |
|----|----------------------------|-----------------------------------------|
| 1  | 1                          | 1                                       |
| 2  | 1                          | 3                                       |
| 3  | 1                          | 9                                       |
| 4  | 2                          | 3                                       |


Anyways - I see that this Sprint is nearly over, so we should maybe just merge it and change it later...

Reply:
This table looks very good, and we can map it directly in the DB, I'm gonna implement it in the next sprint  :)